website/docs/best-practices/FileDownloads.md

```diff
@@ -0,0 +1,107 @@
+---
+id: file-download
+title: File Download
+---
+
+# File Download
+
+When automating file downloads in web testing, it's essential to handle them consistently across different browsers to ensure reliable test execution.
+
+Here, we provide best practices for file downloads and demonstrate how to configure download directories for **Google Chrome**, **Mozilla Firefox**, and **Microsoft Edge**.
+
+## Download Paths
+
+**Hardcoding** download paths in test scripts can lead to maintenance issues and portability problems. Utilize **relative paths** for download directories to ensure portability and compatibility across different environments.
+
+```javascript
+// 👎
+// Hardcoded download path
+const downloadPath = '/path/to/downloads';
+
+// 👍
+// Relative download path
+const downloadPath = path.join(__dirname, 'downloads');
+```
+
+## Wait Strategies
+
+Failing to implement proper wait strategies can lead to race conditions or unreliable tests, especially for download completion. Implement **explicit** wait strategies to wait for file downloads to complete, ensuring synchronization between test steps.
+
+```javascript
+// 👎
+// No explicit wait for download completion
+await browser.pause(5000);
+
+// 👍
+// Wait for file download completion
+await waitUntil(async ()=> await fs.existsSync(downloadPath), 5000);
+```
+
+## Configuring Download Directories With WebdriverIO
+
+To override file download behavior for **Google Chrome**, **Mozilla Firefox**, and **Microsoft Edge**, provide the download directory in the WebDriverIO capabilities:
+
+```javascript
+// Google Chrome
+capabilities: {
+    browserName: 'chrome',
+    'goog:chromeOptions': {
+        prefs: {
+            "download.default_directory": downloadPath
+        }
+    }
+}
+
+// Mozilla Firefox
+capabilities: {
+    browserName: 'firefox',
+    'moz:firefoxOptions': {
+        prefs: {
+            'browser.download.dir': downloadPath,
+            'browser.download.folderList': 2,
+            'browser.download.manager.showWhenStarting': false,
+            'browser.helperApps.neverAsk.saveToDisk': '*/*'
+        }
+    }
+}
+
+// Microsoft Edge
+capabilities: {
+    browserName: 'msedge',
+    'ms:edgeOptions': {
+        prefs: {
+            "download.default_directory": downloadPath
+        }
+    }
+}
+```
```

Comment:
Let's use tables for this one
```suggestion
<Tabs
  defaultValue="chrome"
  values={[
    {label: 'Chrome', value: 'chrome'},
    {label: 'Firefox', value: 'firefox'},
    {label: 'Microsoft Edge', value: 'edge'},
  ]
}>
<TabItem value="chrome">

```js
capabilities: {
    browserName: 'chrome',
    'goog:chromeOptions': {
        prefs: {
            'download.default_directory': downloadPath
        }
    }
}
```

</TabItem>
<TabItem value="firefox">

```js
capabilities: {
    browserName: 'firefox',
    'moz:firefoxOptions': {
        prefs: {
            'browser.download.dir': downloadPath,
            'browser.download.folderList': 2,
            'browser.download.manager.showWhenStarting': false,
            'browser.helperApps.neverAsk.saveToDisk': '*/*'
        }
    }
}
```

</TabItem>
<TabItem value="edge">

```js
capabilities: {
    browserName: 'msedge',
    'ms:edgeOptions': {
        prefs: {
            "download.default_directory": downloadPath
        }
    }
}
```

</TabItem>
</Tabs>
```